doc: explain path.format expected properties

eversojk opened the pull request:
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `make -j8 test` (UNIX) or `vcbuild test nosign` (Windows) pass with
  this change (including linting)?
- [X] Is the commit message formatted according to [CONTRIBUTING.md][0]?
- [doesn't fix a bug] If this change fixes a bug (or a performance problem), is a regression
  test (or a benchmark) included?
- [X] Is a documentation update included (if this change modifies
  existing APIs, or introduces new ones)?

### Affected core subsystem(s)

path

### Description of change

explain path.format expected properties

fixes #5746

benjamingr commented:
Thanks, please wrap lines at 80 characters if that's not too much work. It could also be be nice to expand a little here.

eversojk commented:
I'll fix the long line, thanks for pointing that out. I was going to expand further with more examples in a different PR fixing https://github.com/nodejs/node/issues/5747. Would adding more examples in a separate PR be preferred?

benjamingr commented:
I think one PR is fine but @Trott is more experienced here than I.

eversojk commented:
Ok, I'll wait until he chimes in before adding the additional examples.

jasnell commented:
One PR is fine as long as the changes are all relevant to the same issue. There can be multiple commits if necessary.

Trott commented:
One PR is fine in this case, but two is also fine, and since you already have it as two, maybe leave it as that. As this is your first commit (right?), it might be best to keep it small anyway so we can focus on process and not endless nits on grammar and style.

Trott commented:
LGTM. If you can squash the two commits down to one and force push, that would be great. If not, whoever lands this can do it.

Trott commented on doc/api/path.markdown:
```diff
@@ -88,9 +88,9 @@ path.extname('.index')
-If `pathObject` has all expected properties, the returned string will be a
-concatenation of the `dir` property, the platform-dependent path separator, and
-the `base` property.
+If `pathObject` has the `dir` and `base` properties, the returned string will
```
Nit: Might be better as "has `dir` and `base` properties" (no "the").

eversojk commented:
@Trott I've added your suggestion and squashed my commits into one and should be good to go. I'll add the extra examples on https://github.com/nodejs/node/issues/5747.

Trott commented:
Still LGTM. 

Nit: The body of the commit message should probably be a sentence explaining the change rather than the first line of the commits that we're squashed.

benjamingr commented:
Thanks for the contribution @eversojk ! Landed in https://github.com/nodejs/node/commit/d16d9047eaa9f50d5dea5d7d42f6ae5b4369fe8b

eversojk commented:
@benjamingr Thanks! Also thanks for your input @jasnell @Trott
